zkp/circuits/anon_enc_nullifier_kyc.circom

```diff
@@ -0,0 +1,156 @@
+  checkOutputHashes.values <== outputValues;
+  checkOutputHashes.salts <== outputSalts;
+  checkOutputHashes.ownerPublicKeys <== outputOwnerPublicKeys;
+
+  component checkNullifiers = CheckNullifiers(nInputs);
+  checkNullifiers.nullifiers <== nullifiers;
+  checkNullifiers.values <== inputValues;
+  checkNullifiers.salts <== inputSalts;
+  checkNullifiers.ownerPrivateKey <== inputOwnerPrivateKey;
+
+  component checkSum = CheckSum(nInputs, nOutputs);
+  checkSum.inputValues <== inputValues;
+  checkSum.outputValues <== outputValues;
+
+  // With the above steps, we demonstrated that the nullifiers
+  // are securely bound to the input commitments. Now we need to
+  // demonstrate that the input commitments belong to the Sparse
+  // Merkle Tree with the root `root`.
+  component checkUTXOSMTProof = CheckSMTProof(nInputs, nUTXOSMTLevels);
+  checkUTXOSMTProof.root <== utxosRoot;
+  checkUTXOSMTProof.merkleProof <== utxosMerkleProof;
+  checkUTXOSMTProof.enabled <== enabled;
+  checkUTXOSMTProof.leafNodeIndexes <== inputCommitments;
+
+  // Then, we need to check that the owner public keys
+  // for the inputs and outputs are included in the identities
+  // Sparse Merkle Tree with the root `identitiesRoot`.
+  var ownerPublicKeyHashes[nOutputs + 1];
+  component hash1 = Poseidon(2);
+  hash1.inputs[0] <== inputOwnerPublicKey[0];
+  hash1.inputs[1] <== inputOwnerPublicKey[1];
+  ownerPublicKeyHashes[0] = hash1.out;
+
+  component hashes[nOutputs];
+  var identitiesMTPCheckEnabled[nOutputs + 1];
+  identitiesMTPCheckEnabled[0] = 1;
+  for (var i = 0; i < nOutputs; i++) {
+    hashes[i] = Poseidon(2);
+    hashes[i].inputs[0] <== outputOwnerPublicKeys[i][0];
+    hashes[i].inputs[1] <== outputOwnerPublicKeys[i][1];
+    ownerPublicKeyHashes[i+1] = hashes[i].out;
+    identitiesMTPCheckEnabled[i+1] = 1;
+  }
+
+  component checkIdentitiesSMTProof = CheckSMTProof(nOutputs + 1, nIdentitiesSMTLevels);
+  checkIdentitiesSMTProof.root <== identitiesRoot;
+  checkIdentitiesSMTProof.merkleProof <== identitiesMerkleProof;
+  checkIdentitiesSMTProof.enabled <== identitiesMTPCheckEnabled;
+  checkIdentitiesSMTProof.leafNodeIndexes <== ownerPublicKeyHashes;
+
+  // generate shared secret
+  var sharedSecret[2];
+  component ecdh = Ecdh();
+  ecdh.privKey <== inputOwnerPrivateKey;
+  // our circuit requires that the output UTXO for the receiver must be the first in the array
+  ecdh.pubKey[0] <== outputOwnerPublicKeys[0][0];
+  ecdh.pubKey[1] <== outputOwnerPublicKeys[0][1];
+  sharedSecret[0] = ecdh.sharedKey[0];
+  sharedSecret[1] = ecdh.sharedKey[1];
+
+  // encrypt the value for the receiver
+  component encrypt = SymmetricEncrypt(2);
+  // our circuit requires that the output UTXO for the receiver must be the first in the array
+  encrypt.plainText[0] <== outputValues[0];
+  encrypt.plainText[1] <== outputSalts[0];
+  encrypt.key <== sharedSecret;
+  encrypt.nonce <== encryptionNonce;
+  encrypt.cipherText[0] --> cipherText[0];
```

Comment:
the 4 assignments here (using `-->`) should all be changed to assign and check with `==>`.

would you mind doing the same for the other circuits using encryption? thanks!